warp/src/raygun/mod.rs

```diff
@@ -548,41 +548,54 @@ impl Default for ConversationSettings {
 #[repr(C)]
 pub struct DirectConversationSettings {}
 
-#[derive(Default, Debug, Copy, Clone, Serialize, Deserialize, PartialEq, Eq, Hash, Display)]
-#[display(
-    fmt = "Everyone can add participants: {}, Everyone can change name: {}",
-    "if self.members_can_add_participants {\"✅\"} else {\"❌\"}",
-    "if self.members_can_change_name {\"✅\"} else {\"❌\"}"
-)]
+#[derive(Default, Debug, Clone, Serialize, Deserialize, PartialEq, Eq, Display)]
+#[display(fmt = "Roles: {:?}", "self.roles")]
 #[repr(C)]
 pub struct GroupSettings {
-    // Everyone can add participants, if set to `true``.
-    #[serde(default)]
-    members_can_add_participants: bool,
-    // Everyone can change the name of the group.
+    // Roles to manage permissions. Key is the unique name of the role.
     #[serde(default)]
-    members_can_change_name: bool,
+    roles: IndexMap<String, Role>,
+}
+
+#[derive(Clone, Eq, PartialEq, Serialize, Deserialize, Debug)]
+pub struct Role {
+    // Participants assigned to this role
+    participants: Vec<DID>,
+    // Permissions of this role
+    permissions: Vec<Permission>,
+}
+
+#[derive(Clone, Eq, PartialEq, Serialize, Deserialize, Debug)]
+pub enum Permission {
+    All,
+    SetRoles,
+    ChangeGroupName,
+    AddParticipantsToGroup,
```

Comment:
Might want to split this into two or maybe have separate enumerator to indicate if the permission can add or remove for any explicit type (though when it comes to exporting this to wasm, I can imagine that being a pain unless we change `Permission` to a struct and have a functions to provide the inner indication. Just something to think about)

Reply:
for group permissions, keeping it to a simple enum, since there will only be a few permissions. However, communities will have more complex permissions, so it's worth considering there.